src/RoomsSDKAdapter.js

```diff
@@ -36,42 +37,67 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
     };
   }
 
+  /**
+   * Create an observable that subscribes to room change events emitted via websocket and
+   * emits the changed room object.
+   *
+   * @private
+   * @param {string} ID ID of room to get
+   * @returns {Observable.<Room>} Observable that emits changes to the room
+   * @memberof RoomsSDKAdapter
+   */
+  addRoomChangeListener(ID) {
+    const roomChange$ = fromEvent(this.datasource.rooms, ROOM_UPDATED_EVENT);
+
+    return roomChange$.pipe(
+      // Is the room change event for our subscribed room?
+      filter((event) => event.data.id === ID),
+      // Event data doesn't have the room data in it, so we need to fetch manually
+      mergeMap(async () => this.fetchRoom(ID))
```

Comment:
Why are you using `mergeMap` with an `async` callback here, instead of using `from` to convert the promise to observable?